### PR TITLE
[CAMERA] Multiple camera modes: LOCKED, FREE, and SUBJECTIVE

### PR DESCRIPTION
**Pull request by 4quet**
Hold right click to look around
Tab to switch between view modes (FREE or SUBJECTIVE)
WASD to move left/right and forward/backward
Space to move up (in FREE mode only)
Left ctrl to move down (in FREE mode only)

### CODE REVIEW

File: XV/Assets/Scripts/Utils/CameraHandler.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class CameraHandler : MonoBehaviour {
+
+	public enum Mode { FREE, SUBJECTIVE, LOCKED }
+
+	[SerializeField] private float StandardSpeed;
```
Comment (banthony42) [inline XV/Assets/Scripts/Utils/CameraHandler.cs]: [SerializeField]
private float standardSpeed;

(Endline)

File: XV/Assets/Scripts/Utils/CameraHandler.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+			if (!mIsRepositioning) {
+				ApplyMovement();
+			}
+			ApplyRotation();
+		}
+    }
+
+	private void ApplyMovement()
+	{
+		bool fastMode = Input.GetKey(KeyCode.LeftShift);
```
Comment (banthony42) [inline XV/Assets/Scripts/Utils/CameraHandler.cs]: Var local = lVar
Var parameter = iVar
Var to return = oVar

Var attribut = mVar
static Var  = sVar

enum = VAR
const = VAR

File: XV/Assets/Scripts/Utils/CameraHandler.cs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+		float xSpeed = Input.GetAxis("Horizontal") * baseSpeed * Time.deltaTime;
+		float zSpeed = Input.GetAxis("Vertical") * baseSpeed * Time.deltaTime;
+
+		Vector3 xAxisMovement = transform.right * xSpeed;
+		Vector3 zAxisMovement = transform.forward * zSpeed;
+
+		// Freeze Y axis movement if in subjective mode
+		if (CameraMode == Mode.SUBJECTIVE) {
+			zAxisMovement.y = 0f;
```
Comment (banthony42) [inline XV/Assets/Scripts/Utils/CameraHandler.cs]: 0F

File: XV/Assets/Scripts/Utils/CameraHandler.cs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+	private void SetSubjectiveMode()
+	{
+		mMode = Mode.SUBJECTIVE;
+		Cursor.visible = false;
+		Cursor.lockState = CursorLockMode.Locked;
+		StartCoroutine(SetSubjectivePosition(new Vector3(transform.position.x, 1f, transform.position.z)));
+	}
+
+	private IEnumerator SetSubjectivePosition(Vector3 iTarget)
```
Comment (banthony42) [inline XV/Assets/Scripts/Utils/CameraHandler.cs]: Add a suffix to the coroutine name:
private IEnumerator SetSubjectivePositionAsync();

File: XV/Assets/Scripts/Utils/CameraHandler.cs [hunk 5]
```diff
@@ -0,0 +1,146 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class CameraHandler : MonoBehaviour {
```
Comment (banthony42) [inline XV/Assets/Scripts/Utils/CameraHandler.cs]: Make your class sealed if it doesn't has to be extended

**Review by banthony42 — CHANGES_REQUESTED** [review]
See comments

**Review by VelocityPulse — APPROVED** [review]
Checked

**Review by banthony42 — APPROVED** [review]
Checked.

**Pull request merged**